packages/hardhat/contracts/YourContract.sol

```diff
@@ -132,8 +134,8 @@ contract YourContract is Ownable {
         uint256[] calldata _caps,
         address[] calldata _optionalTokenAddresses
     ) public onlyOwner {
-        require(_builders.length == _caps.length && _builders.length == _optionalTokenAddresses.length, "Lengths are not equal");
         uint256 length = _builders.length; 
+        if (length != _caps.length || length != _optionalTokenAddresses.length) revert ArrayLengthMismatch();
```

Comment:
we can get this in good catch

Reply:
Cool, thanks!

Don't forget to include the changes in the `for-loop` [below your comment](https://github.com/lokithe5th/hacker-houses-streams/blob/0152f928ad72b43c4034e0b79fcf9923b0887740/packages/hardhat/contracts/YourContract.sol#L141-L148). It just moves the event outside of the loop and adds a custom emission for Batches. This leads to a significant gas saving of ~ 250 000 gas for an array of 10 builders added.